PR: https://github.com/odoo/odoo/pull/78730

From: bc772f4784abb0b7660169838b0be487445a5122
From: jbw
Date: 2021-11-25 09:13:47

Structural Changes: 8
Total Changes: 20

[IMP]  account: track lock dates changes

Allow log date changes tracking by adding chatter on company and tracking lock date fields.

closes odoo/odoo#78730

Task: 2651834
Related: odoo/enterprise#22479
Signed-off-by: Laurent Smet <las@openerp.com>

================================= pseudo patch: =================================

--- a/addons/account/models/company.py
+++ b/addons/account/models/company.py
@@ -36,14 +36,26 @@ DASHBOARD_ONBOARDING_STATES = ONBOARDING_STEP_STATES + [('closed', 'Closed')]
 
 
 class ResCompany(models.Model):
-    _inherit = "res.company"
+    _name = "res.company"
+    _inherit = ["res.company", "mail.thread"]
 
     #TODO check all the options/fields are in the views (settings + company form view)
     fiscalyear_last_day = fields.Integer(default=31, required=True)
     fiscalyear_last_month = fields.Selection(MONTH_SELECTION, default='12', required=True)
-    period_lock_date = fields.Date(string="Lock Date for Non-Advisers", help="Only users with the 'Adviser' role can edit accounts prior to and inclusive of this date. Use it for period locking inside an open fiscal year, for example.")
-    fiscalyear_lock_date = fields.Date(string="Lock Date", help="No users, including Advisers, can edit accounts prior to and inclusive of this date. Use it for fiscal year locking for example.")
-    tax_lock_date = fields.Date("Tax Lock Date", help="No users can edit journal entries related to a tax prior and inclusive of this date.")
+    period_lock_date = fields.Date(
+        string="Journals Entries Lock Date",
+        tracking=True,
+        help="Only users with the 'Adviser' role can edit accounts prior to and inclusive of this"
+             " date. Use it for period locking inside an open fiscal year, for example.")
+    fiscalyear_lock_date = fields.Date(
+        string="All Users Lock Date",
+        tracking=True,
+        help="No users, including Advisers, can edit accounts prior to and inclusive of this date."
+             " Use it for fiscal year locking for example.")
+    tax_lock_date = fields.Date(
+        string="Tax Return Lock Date",
+        tracking=True,
+        help="No users can edit journal entries related to a tax prior and inclusive of this date.")
     transfer_account_id = fields.Many2one('account.account',
         domain=lambda self: [('reconcile', '=', True), ('user_type_id.id', '=', self.env.ref('account.data_account_type_current_assets').id), ('deprecated', '=', False)], string="Inter-Banks Transfer Account", help="Intermediary account used when moving money from a liquidity account to another")
     expects_chart_of_accounts = fields.Boolean(string='Expects a Chart of Accounts', default=True)
